### DISCUSSION TITLE
BrnMenuTriggerDirective causing some problems in my tests

### DISCUSSION THREAD

**Discussion by elfoxus** [spartan-ng/spartan#300] [Q&A]
Hi. I've got a problem with unit testing my components. I am using BrnMenuTriggerDirective in header of my page. I've wanted to write some unit tests for my view, however my tests will not start, because I've got problem with this directive, as you can see below:

`NG0311: Directive _CdkMenuTrigger does not have an input with a public name of brnMenuTriggerFor.`

I have copied the entire code of the directive into my tests and replaced it with overrideComponent method of TestBed. Suprisingly it works. What might be the cause of it? Any ideas?

**Comment by goetzrobin** [comment]
Hey there! Can you share a reproduction in stack blitz

**Reply by elfoxus** [reply]
Sure, I will try to create repro
